src/box/sql/mem.c

```diff
@@ -2868,6 +2870,36 @@ port_lua_get_vdbemem(struct port *base, uint32_t *size)
 					 field.sval.len) != 0)
 				goto error;
 			break;
+		case MP_EXT: {
+			assert(field.ext_type == MP_UUID ||
+			       field.ext_type == MP_DECIMAL);
+			char *buf;
+			uint32_t size;
+			uint32_t svp = region_used(&fiber()->gc);
+			if (field.ext_type == MP_UUID) {
+				size = mp_sizeof_uuid();
+				buf = region_alloc(&fiber()->gc, size);
+				if (buf == NULL) {
+					diag_set(OutOfMemory, size,
+						 "region_alloc", "buf");
+					goto error;
+				}
+				mp_encode_uuid(buf, field.uuidval);
+			} else {
+				size = mp_sizeof_decimal(field.decval);
+				buf = region_alloc(&fiber()->gc, size);
+				if (buf == NULL) {
+					diag_set(OutOfMemory, size,
+						 "region_alloc", "buf");
+					goto error;
+				}
+				mp_encode_decimal(buf, field.decval);
+			}
+			if (mem_copy_bin(&val[i], buf, size) != 0)
+				goto error;
+			region_truncate(&fiber()->gc, svp);
```

Comment:
If mem_copy_bin fails, region doesn't get truncation.

Reply:
It does not matter, since in this case region will be just cleared by VDBE during destruction.

Reply:
Ok, I fixed this.

Also, I'd like to say that this part with region will soon be removed since UUID and DECIMAL will be part of union u of struct Mem.